components/scream/src/dynamics/homme/atmosphere_dynamics.cpp

```diff
@@ -516,7 +516,7 @@ void HommeDynamics::initialize_impl (const util::TimeStamp& /* t0 */)
       m_ref_grid_fields.at("w_int").get_header().get_tracking().get_providers().size()==1,
       "Error! Someone other than Dynamics is trying to update the vertical velocity.\n");
 
-  auto positivity_check = std::make_shared<FieldPositivityCheck<Real> >();
+  auto positivity_check = std::make_shared<FieldPositivityCheck<Real> >(0.0);
```

Comment:
We should add a comment here, that clarifies that we are repairing the field if needed.

Also, we need to make sure we are not clipping stuff that is not morally zero. We know that _right now_ the negative values are benign, but once this clipping goes in, we may not spot others.

I think we should do something better here. I see 2 options:

1) make the check `f >-tol`, with tol not too big, to make it menaingful, and not too small to make it throw with not-so-small benign neg values. Do not allow repair. Then add a second check that does simple positivity check, and allow repair.
2) implement a more robust check, which enforces `f(icol,:) > -10*eps*abs(max(f(icol,:)))`, and also can repair.

The latter is more robust, and we should be the long term solution. Even if we manually rolled it in the atm dyn interface (without writing a FPC class). However, the former is probably faster to implement. Notice that the 1st is "safe", since checks are internally stored in the order in which they are added, so we can always enforce that a FPC is run before another, by adding it first (if FPC's were stored as an ordered container, such as `std::set`, this would not be possible).

tl;dr: can we add a FieldWithinIntervalCheck _before_ the positivity check, with lower bound `-1e-10`, upper bound `std;:numeric_limits<Real>::max()`, and `can_repair=false`?

Reply:
Good idea.  I like option 2, but it seems like option 1 would be the fastest solution for now.  I can go with that but put in a comment about how we would like to implement option 2 eventually.